sklearn/tests/test_impute.py

```diff
@@ -1009,9 +1025,22 @@ def test_missing_indicator_sparse_param(arr_type, missing_values,
 
     indicator = MissingIndicator(missing_values=missing_values,
                                  sparse=param_sparse)
+
+    if param_sparse is True and missing_values == 0:
```

Comment:
```suggestion
    if param_sparse and missing_values == 0:
```

Reply:
wouldn't it also cover the case when "param_sparse=='auto'", which should be an allowed case